host/sgx/sgxsign.c

```diff
@@ -664,3 +737,95 @@ oe_result_t oe_sgx_sign_enclave(
 
     return result;
 }
+
+oe_result_t oe_sgx_get_sigstruct_digest(
+    const OE_SHA256* mrenclave,
+    uint64_t attributes,
+    uint16_t product_id,
+    uint16_t security_version,
+    OE_SHA256* digest)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    sgx_sigstruct_t sigstruct;
+
+    if (digest)
```

Comment:
Consider keep this check and remove the same check below (`!digest)` or remove this check and put the `memset`  after the checks of parameters.

Reply:
This replicates the pattern that Mike put in place to always return zeroed out buffers on failed function calls, which is why it precedes the invalid parameter check.

Reply:
Thanks for the clarification!